server/src/routes/Community.ts

```diff
@@ -0,0 +1,13 @@
+import express from 'express';
+import * as communityController from '../controllers/CommunityController';
+import { validateToken } from '../middlewares';
+
+const router = express.Router();
+
+router.get(
+  '/all-communities',
```

Comment:
This isn't a big deal just kind of a naming thing with restful apis. This route is already under "api/community" even though its singular in name, that route would actually provide a list of all of them. Meanwhile, a route like "api/community/{community_id}" would provide information of a specific community based on the id. This hits the 2 usual requirements of everything and something specific.

I've seen both singular and plural used for these base routes, "users" instead of "user". So if that seems more natural to have this pattern we can change to that in another pr or leave it as is.

TLDR: "api/communities/" would be a route that returns a list of all of them and "api/communities/{id}" would be specific, instead of having a route called "all-communities".

Reply:
That makes a lot of sense, thank you for the tip. I think for the time being we can just leave this as-is and revisit it later.